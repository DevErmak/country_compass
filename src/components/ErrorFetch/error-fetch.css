.error-fetch-container {
  text-align: center;
  width: 100vw;
  height: 100vh;
  border-radius: 10px;
  display: flex;
  direction: column;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
  padding: 50px;
  box-sizing: border-box;
}
.info-error-fetch {
  color: #8d4747;
  text-align: center;
  width: 250px;
  height: 250px;
  border-radius: 10px;
  background-color: #fff;
  display: flex;
  direction: column;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
  padding: 50px;
}
.try-again {
  text-align: center;
  width: 133px;
  height: 41px;
  flex-shrink: 0;
  border: none;
  border-radius: 15px;
  background: #d9d9d9;
  padding: 6px 30px 7px 29px;
  font-size: 20px;
  font-weight: 600;
  margin-top: 33.33px;
}
.try-again:hover {
  background: #8d4747;
  color: #000;
  transition: 0.5s;
}
