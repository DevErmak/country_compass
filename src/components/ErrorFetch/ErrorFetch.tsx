import React from 'react';
import { useNavigate } from 'react-router-dom';
import './error-fetch.css';
type Props = { infoError: string };

export default function ErrorFetch({ infoError }: Props) {
  const navigate = useNavigate();

  return (
    <div className="error-fetch-container">
      <div className="info-error-fetch">
        <div>{infoError}.</div>
        <div className="try-again" onClick={() => navigate('/')}>
          Try again
        </div>
      </div>
    </div>
  );
}
