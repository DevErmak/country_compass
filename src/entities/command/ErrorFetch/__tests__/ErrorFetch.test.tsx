import renderer from 'react-test-renderer';
import ErrorFetch from '../ErrorFetch';
import { renderComponent } from '../../../shared/lib/test-helpers';

it('renders correctly ErrorFetch', () => {
  const { container } = renderComponent({
    Component: ErrorFetch,
    props: { infoError: '' },
  });
  expect(container).toMatchSnapshot();
});
