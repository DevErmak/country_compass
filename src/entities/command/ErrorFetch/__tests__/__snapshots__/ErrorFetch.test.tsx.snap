// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly ErrorFetch 1`] = `
<div>
  <div
    class="error-fetch-container"
  >
    <div
      class="info-error-fetch"
    >
      <div>
        .
      </div>
      <div
        class="try-again"
      >
        Try again
      </div>
    </div>
  </div>
  <div
    id="portal"
  />
</div>
`;
